src/dataframe.rs

```diff
@@ -70,8 +73,9 @@ impl PyDataFrame {
         if let Ok(key) = key.extract::<PyBackedStr>() {
             // df[col]
             self.select_columns(vec![key])
-        } else if let Ok(tuple) = key.extract::<&PyTuple>() {
+        } else if let Ok(tuple) = key.extract::<Py<PyTuple>>() {
             // df[col1, col2, col3]
+            let tuple = tuple.bind(key.py());
```

Comment:
```suggestion
        } else if let Ok(tuple) = key.downcast::<PyTuple>() {
            // df[col1, col2, col3]
```

Reply:
What's the difference between extract() and downcast()? I was trying to figure it out with chatgpt and failed 😆

Reply:
Ref:

https://pyo3.rs/v0.22.2/performance.html?highlight=downcast#extract-versus-downcast